servant-auth-server/test/Servant/Auth/ServerSpec.hs

```diff
@@ -336,31 +338,33 @@ jwtCfg = (defaultJWTSettings theKey) { audienceMatches = \x ->
     if x == "boo" then DoesNotMatch else Matches }
 
 instance FromBasicAuthData User where
-  fromBasicAuthData (BasicAuthData usr pwd) _
-    = return $ if usr == "ali" && pwd == "Open sesame"
-      then Authenticated $ User "ali" "ali@the-thieves-den.com"
-      else Indefinite
+  fromBasicAuthData (BasicAuthData usr pwd) _ = return $
+    if usr /= "ali"
+    then Left BasicAuthNoSuchUser
+    else if pwd /= "Open sesame"
+      then Left BasicAuthBadPassword
+      else Right $ User "ali" "ali@the-thieves-den.com"
 
 -- Could be anything, really, but since this is already in the cfg we don't
 -- have to add it
 type instance BasicAuthCfg = JWK
 
 -- | Takes a proxy parameter indicating which authentication systems to enable.
-app :: AreAuths auths '[CookieSettings, JWTSettings, JWK] User
+app :: (AreAuths auths '[CookieSettings, JWTSettings, JWK] User
+       , Show (ErrorList (AuthsToErrors auths)))
   => Proxy (API auths) -> Application
 app api = serveWithContext api ctx server
   where
     ctx = cookieCfg :. jwtCfg :. theKey :. EmptyContext
 
 
-server :: Server (API auths)
+server :: Show (ErrorList (AuthsToErrors auths)) => Server (API auths)
```

Comment:
Could this be abstracted into a small type synonym, so that we can instead say something like:
```
server :: auths `AuthErrorsAre` Show => Server (API auths)
```
(I just imagine it's going to be pretty common to require that e.g. they be `ToJSON` etc. and it'd be nice to not surface `ErrorList` and `AuthsToErrors` too much, I think)

Reply:
I'm not sure this is necessary in the light of your suggested `ErrorList` modification. We end up with just `Show (ErrorList auths)` which I think is manageable.

Reply:
Indeed.